src/SW_Weather.c

```diff
@@ -1177,13 +1656,103 @@ void _read_weather_hist(
+                                                      weathInput[dailyInputIndices[REL_HUMID_MIN]]) / 2;
+
+            } else if(dailyInputFlags[REL_HUMID]) {
+                yearWeather->r_humidity_daily[doy] = weathInput[dailyInputIndices[REL_HUMID]];
+
+            } else if(dailyInputFlags[SPEC_HUMID]) {
+                // Specific Humidity (Bolton 1980)
+                es = (6.112 * exp(17.67 * yearWeather->temp_avg[doy]) /
+                     (yearWeather->temp_avg[doy] + 243.5));
+
+                e = (weathInput[dailyInputIndices[SPEC_HUMID]] * 1013.25) /
+                    (.378 * weathInput[dailyInputIndices[SPEC_HUMID]] + .622);
+
+                relHum = e / es;
+                relHum = max(0., relHum);
+
+                yearWeather->r_humidity_daily[doy] = min(100., relHum);
+
+                // Check if a calculation of actual vapor pressure occurred with
+                // bad temperature values (SW_MISSING)
+                if(missing(yearWeather->temp_avg[doy])) {
+
+                    yearWeather->r_humidity_daily[doy] = SW_MISSING;
+                }
+
+            }
+
+            // Deal with actual vapor pressure
+            if(dailyInputFlags[ACTUAL_VP]){
+
+                yearWeather->actualVaporPressure[doy] = weathInput[dailyInputIndices[ACTUAL_VP]];
+
+            } else if(dailyInputFlags[TEMP_DEWPOINT]) {
+                    yearWeather->actualVaporPressure[doy] =
+                                    actualVaporPressure3(weathInput[dailyInputIndices[TEMP_DEWPOINT]]);
+
+            } else if(hasMaxMinTemp && hasMaxMinRelHumid) {
+                yearWeather->actualVaporPressure[doy] =
+                                actualVaporPressure2(weathInput[dailyInputIndices[REL_HUMID_MAX]],
+                                                     weathInput[dailyInputIndices[REL_HUMID_MIN]],
+                                                     weathInput[dailyInputIndices[TEMP_MAX]],
+                                                     weathInput[dailyInputIndices[TEMP_MIN]]);
+
+                // Check if a calculation of relative humidity occurred with
+                // bad temperature values (SW_MISSING)
+                if(missing(yearWeather->temp_max[doy]) || missing(yearWeather->temp_min[doy])) {
+
+                    yearWeather->r_humidity_daily[doy] = SW_MISSING;
+                }
+
+            } else if(dailyInputFlags[REL_HUMID] || dailyInputFlags[SPEC_HUMID]) {
+                yearWeather->actualVaporPressure[doy] =
+                             actualVaporPressure1(yearWeather->r_humidity_daily[doy],
+                                                  yearWeather->temp_avg[doy]);
+            }
+
+            // Check if a better calculation of relative humidity is available
+            // (using dewpoint temperature or the daily actual vapor pressure input value)
+            if(dailyInputFlags[ACTUAL_VP] || dailyInputFlags[TEMP_DEWPOINT]) {

```

Comment:
Calculate `yearWeather->r_humidity_daily[doy]` only from vapor pressure or dewpoint temperature if not otherwise provided as input, i.e., if missing